src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/ParameterCapturingLogger.cs

```diff
@@ -95,8 +106,6 @@ public void Dispose()
                 return;
             }
 
```

Comment:
> Shouldn't we still call `.CompleteAdding()` here so that `while (_messages.TryTake(out (string format, string[] args) entry, Timeout.InfiniteTimeSpan))` returns?

Good catch, `Dispose` does not interrupt TryTake but does prevent another `TryTake` from occuring.